git-hooks/commit/errors.js

```diff
@@ -0,0 +1,86 @@
+const TYPES = require('./types')
+const { format } = require('../util/formatting')
+
+/* STYLES */
+const h1 = format({ fg: 'red', bold: true })
+const h2 = format({ fg: 'cyan', bold: true })
+const code = format({ fg: 'gray' })
+
+
+/* SNIPPETS  */
+const SYNTAX = '<type>[(<scope>)][!]: <summary>'
+const SYNTAX_NO_SCOPE = '<type>[!]: <summary>'
+const TYPES_INLINE = Object.keys(TYPES).join(', ')
+
+
+/* ERROR FUNCTIONS */
+
+const incorrectFormat = (txtMsg) => console.error(`
+${h1('ERROR: Submitted commit message header does not match expected format!')}
+
+${h2('Submitted')}
+    ${txtMsg}
+
+${h2('Syntax')}
+    ${code(SYNTAX)}
+`)
+
+
+const unknownType = (header, type) => console.error(`
+${h1('ERROR: Submitted commit message header has unknown commit type!')}
+
+${h2('Submitted Header')}
+    ${header}
+
+${h2('Parsed Type')}
+    ${type}
+
+${h2('Valid Types')}
+    ${TYPES_INLINE}
+
+${h2('Syntax')}
+    ${code(SYNTAX)}
+`)
+
+
+const unusableContent = (msgContent) => console.error(`
+${h1('ERROR: Submitted commit message contains no useful content!')}
+
+Please make sure that your commit message has at least one
+line of non-comment text.
+
+${h2('Syntax')}
+    ${code(SYNTAX)}
+
+    ${code('[<body>]')}
+
+    ${code('[<footer>]')}
+
+${h2('Submitted Message Content')}
+${code(msgContent)}
+`)
```

Comment:
**Short Example (empty commit message)**
![CleanShot 2019-10-16 at 17 15 27](https://user-images.githubusercontent.com/545605/66963208-c24a6100-f038-11e9-87b9-2cef64bb550c.png)

**Long Example (comments and whitespace only)**
![CleanShot 2019-10-16 at 17 16 18](https://user-images.githubusercontent.com/545605/66963269-e1e18980-f038-11e9-9e37-ed41591d493f.png)